src/enway_reach_rs_driver/reach_rs_driver.py

```diff
@@ -82,65 +83,71 @@ def receivesFixes(self):
-        elif self.isConnected():
+        elif self.connected_:
             stat.summary(diagnostic_msgs.msg.DiagnosticStatus.WARN, 'Reach RS driver is connected but has no fix')
         else:
             stat.summary(diagnostic_msgs.msg.DiagnosticStatus.ERROR, 'Reach RS driver has a connection problem')
             
-        stat.add("Connection status", self.connectionStatus_)
+        stat.add('Connected', self.connected_)
+        stat.add('Connection status', self.connection_status_)
         
         if self.lastFix_:
             stat.add("Fix status", FixStatus[self.lastFix_.status.status])
             stat.add("Seconds since last fix", (rospy.Time.now()-self.lastFix_.header.stamp).to_sec())
         else:
             stat.add("Fix status", FixStatus[NavSatStatus.STATUS_NO_FIX])
-            stat.add("Seconds since last fix", FixStatus[NavSatStatus.STATUS_NO_FIX])
+            stat.add("Seconds since last fix", '-')
         
     def connectToDevice(self):
-        rospy.loginfo('Connecting to {0}:{1}...'.format(self.host_, self.port_))
-        
-        self.socket_.settimeout(0.2)
+        rospy.loginfo('Connecting to {0}:{1}...'.format(*self.address_))
         
         while not rospy.is_shutdown():
             self.update()
             
+            if self.socket_:
+                self.socket_.close()
+                self.socket_ = None
+
+            self.connected_ = False
+            self.connection_status_ = 'not connected'
+
+            self.socket_ = socket.socket()
+
             try:
-                self.socket_.connect((self.host_, self.port_))
-                self.connectionStatus_ = 'Connected'
+                self.socket_.settimeout(5)
+                self.socket_.connect(self.address_)
+                self.connected_ = True
+                self.connection_status_ = 'connected'
                 rospy.loginfo('Successfully connected to device!')
                 return
             except socket.timeout:
-                self.connectionStatus_ = 'Connect timeout. Retrying...'
+                self.connection_status_ = 'connect timeout'
             except socket.error,  msg:
-                self.connectionStatus_ = 'Connect error: {0}. Retrying...'.format(msg)
+                self.connection_status_ = 'connect error ({0})'.format(msg)
             
         exit()
         
-    def reconnectToDevice(self):
-        rospy.logwarn('Lost connection. Trying to reconnect...')
-        
-        self.connectionStatus_ = 'Not connected'
-        self.socket_.close()
-        self.socket_ = socket.socket()
-        self.connectToDevice()
-        
     def run(self):
         self.connectToDevice()
         
         while not rospy.is_shutdown():
             self.update()
             
             try:
-                self.socket_.settimeout(0.1)
+                self.socket_.settimeout(self.fixTimeout_)
```

Comment:
Since you fixed the names of all other member variables, you could change this one.

Reply:
Good point, I changed the naming.